### PR TITLE
Feature/http server

### PR DESCRIPTION
**Pull request by rnemes**
## Bug / Requirement Description
Added the possibility to access the data of the request received in HTTP Server object.

## Solution description
Introduced a ReceivedRequest class and added seome helper functions

## Checklist:
- [x] Test
- [x] Example (both test_plan.py and .rst)
- [x] Documentation (API)
- [x] News fragment present for release notes
- [x] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

**Comment by M6AI** [comment]
Black formatter needs to be applied.

File: examples/Transports/HTTP/http_basic.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 """Tests HTTP requests between a server and a client."""
-import json
+import json, time
```
Comment (M6AI) [inline examples/Transports/HTTP/http_basic.py:2]: For consistency, please push `import time` onto a new line. Wondering if this import is used. Probably was there for local testing?

**Review by M6AI — CHANGES_REQUESTED** [review]
Added a couple of comments on todos, lets talk offline as well. The example is essentially a functional test, wonder if some unit test would make sense too.

File: testplan/testing/multitest/driver/http/server.py [hunk 2]
```diff
@@ -14,27 +15,124 @@
...
+        ):
+            return json.loads(self.raw_data)
+        else:
+            return None
+
+
+class HTTPResponse:
+    """
+    HTTPResponse containing the status code, headers and content.
+    """
```
Comment (M6AI) [inline testplan/testing/multitest/driver/http/server.py:99]: Looks like a namedtuple to me. Lets discuss.

Comment (rnemes) [reply testplan/testing/multitest/driver/http/server.py:99]: This code was already here, I only moved it upper because HTTPRequestHandler uses it.

**Review by Pyifan — COMMENTED** [review]
This looks pretty good! Well done Robi! I will leave it to you and Marci to figure out any remaining details that might need to be polished.

**Review by M6AI — APPROVED** [review]

**Pull request merged**